Shared/Samples/Download preplanned map area/DownloadPreplannedMapAreaView.Model.swift

```diff
@@ -0,0 +1,311 @@
+        private let temporaryDirectory: URL
+    
+        /// The offline map information.
+        @Published private(set) var offlineMapModels: Result<[OfflineMapModel], Error>?
+        
+        /// All the offline map models or an empty array in the case of an error.
+        private var allOfflineMapModels: [OfflineMapModel] {
+            guard case .success(let models) = offlineMapModels else {
+                return []
+            }
+            return models
+        }
+        
+        /// A Boolean value indicating if the offline content can be deleted.
+        @Published private(set) var canRemoveDownloadedMaps = false
+        
+        init() {
+            // Create temp dsirectory.
+            temporaryDirectory = FileManager.createTemporaryDirectory()
+            
+            // Initializes the online map and offline map task.
+            onlineMap = Map(item: portalItem)
+            offlineMapTask = OfflineMapTask(portalItem: portalItem)
+        }
+        
+        /// Gets the preplanned map areas from the offline map task and creates the
+        /// offline map models.
+        func makeOfflineMapModels() async {
+            self.offlineMapModels = await Result {
+                try await offlineMapTask.preplannedMapAreas
+                    .sorted(using: KeyPathComparator(\.portalItem.title))
+                    .map {
+                        OfflineMapModel(
+                            preplannedMapArea: $0,
+                            offlineMapTask: offlineMapTask,
+                            temporaryDirectory: temporaryDirectory
+                        )
+                    }
+            }
+        }
+        
+        deinit {
+            // Removes the temporary directory
+            try? FileManager.default.removeItem(at: temporaryDirectory)
+        }
+        
+        /// Updates the displayed map based on the given preplanned map area. If the preplanned map
+        /// area is not nil, the preplanned map area will be downloaded if necessary and updates the map
+        /// to the currently selected preplanned map area. If the preplanned map area is nil, then the map
+        /// is set to the online web map.
+        private func selectedMapDidChange(from oldValue: SelectedMap) {
+            switch selectedMap {
+            case .onlineWebMap:
+                offlineMap = nil
+            case .offlineMap(let info):
+                if info.canDownload {
+                    // If we have not yet downloaded or started downloading, then kick off a
+                    // download and reset selection to prevous selection since we have to download
+                    // the offline map.
+                    selectedMap = oldValue
+                    Task { [weak self] in
+                        await info.download()
+                        self?.updateCanRemoveDownloadedMaps()
+                    }
```

Comment:
This unstructured task is fine because when the OfflineMapModels are denoted, they cancel the jobs, which will cause the task to exit.